core/templates/hashfuncs.h

```diff
@@ -201,17 +201,18 @@ static _FORCE_INLINE_ uint32_t hash_fmix32(uint32_t h) {
 static _FORCE_INLINE_ uint32_t hash_murmur3_buffer(const void *key, int length, const uint32_t seed = HASH_MURMUR3_SEED) {
 	// Although not required, this is a random prime number.
 	const uint8_t *data = (const uint8_t *)key;
-	const int nblocks = length / 4;
+	const int nblocks = length / sizeof(uint32_t);
```

Comment:
I'd add a comment here that we might be ignoring trailing bytes, and that this is legacy behavior.

Reply:
This one has always handled the tail in the block below, so there shouldn't be any change.